lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4379,8 +4379,6 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
                    options::OPT_fno_vectorize, EnableVec))
     CmdArgs.push_back("-vectorize-loops");
 
-  } // if (!IsHCCKernelPath)
-
   // -fslp-vectorize is enabled based on the optimization level selected.
   bool EnableSLPVec = shouldEnableVectorizerAtOLevel(Args, true);
```

Comment:
If we change add `&& !IsHCCKernelPath` here, then we would turn off SLP vectorization by default but would still allow it to be turned on if the SLP vectorization compiler switch is explicitly specified?

Reply:
Should we allow it to be turned on if the switch is explicit? But turn off if it is due to O1 or larger?

Reply:
>Should we allow it to be turned on if the switch is explicit?
yes

>But turn off if it is due to O1 or larger?
turn off by default only if HCCKernelPath is in effect

Reply:
1) When I changed to add `&& !IsHCCKernelPath` to bool `EnableSLPVec` initialization:
`bool EnableSLPVec = shouldEnableVectorizerAtOLevel(Args, true) && !IsHCCKernelPath;`
to see if this could remove push_back of `-vectorize-slp`. Seems that with this change, each object will compile with **2 push_backs** of `-vectorize-slp`. 15m8s
2) When we had the change where `if (!IsHCCKernelPath)` surrounds slp-vectorize, we would have **2 push_backs** of `-vectorize-slp`. 15m
3) Without either of these changes, we have **3 push_backs** of `-vectorize-slp`. 16m11.6s
4) I'm not too sure how to have **0 push_backs** other than commenting it out.